### ISSUE TITLE
Replace select2 with something more accessible

### ISSUE THREAD

**Issue by tchak** [betagouv/demarches-simplifiees.fr#4865]
We need to remove `select2`

- depends on jQuery
- not accessible

Replace with maybe https://reacttraining.com/reach-ui/combobox ?

**Comment by Keirua** [comment]
Pour rendre l'autocomplete accessible:
 - soit on prend un autre composant plus accessible, il faut en trouver un et faire les adaptations
 - soit select2 corrige/améliore et c'est top

L'accessibilité est [un sujet chez select2](https://github.com/select2/select2/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility), ils font régulièrement des corrections.

On utilise select2@^4.0.13.  La version 4.1 ajoute des [améliorations d'accessibilité](https://github.com/select2/select2/releases/tag/4.1.0-beta.1), à voir si ça pourrait nous aller:
 - Selected results in the dropdown should now be properly announced to screen readers (#5841)
 - Significant improvements were made to make the selection area accessible (#5842)

**Comment by tchak** [comment]
Nous avons utilisé reach-ui/combobox dans le nouveau module de carto (recherche adresse). Je serai assez curieux de retours à l'occasion.

**Comment by Keirua** [comment]
Les options dont on discutait avec Paul:
 - **facile**: monter de version select2 dès que possible (pour le moment, les évols d'accessibilité sont en beta)
 - **moyen**: remplacer les select par des radiobutton lorsqu'il y a un nombre limité de choix (ce que fait typeform). Semble un très bon quick win.
 - **plus de boulot**: dégager select2 (dernier composant jquery), avoir du react à la volée comme on fait pour la carto pour utiliser un selecteur accessible

J'ai sorti quelques stats sur les nombres de champs dans les dropdown:

|Nombre de champ dans le dropdown | Pourcentage d'utilisation cumulée|
|---------------------------------|----------------------------------|
|3                                | 24%                              |
|4                                | 46%                              |
|5                                | 61%                              |
|6                                | 69%                              |

24% des dropdown contiennent jusqu'à 3 champs, 69% en contiennent jusqu'à 6.
Faudra tester un peu le rendu, mais entre 3 et 6 champs on peut avoir un bon quick win sur l'accessibilité de ce sélecteur

pour sortir ces stats:
```ruby
tdc = TypeDeChamp.includes(:procedure, :drop_down_list).where(type_champ: 'drop_down_list')
tdc_published = tdc.select { |t| t&.procedure&.publiee? }
counts = tdc_published.map { |dd| dd.drop_down_list&.options&.count }
sorted_counts = counts.inject(Hash.new(0)){|p,v| p[v]+=1; p}
y Hash[sorted_counts.sort_by{ |k, v| k || 0 }]
```

**Comment by krichtof** [comment]
Je vais tenter de remplacer les selects par un composant react qui permettra d'avoir un sélecteur accessible

**Comment by tchak** [comment]
Nous avons une PR qui remplace `select2` avec `@reach/combobox` pour les champs `commune`, `regions`, `departements`, `adresse` – #5654 

Il reste :

- [ ] champ `pays` (usager)
- [ ] champ select multiple (usager)
- [ ] envoyer dossier aux instructeurs (instructeur)
- [ ] affecter instructeurs à la démarche (administrateur)
- [ ] selection de colonies affiché pour un dossier (instructeur)

Pour les 4 derniers il faut un multi select avec les "tokens". Si on veut utiliser `@reach/combobox` il y a un exemple ici : [token-input](https://github.com/reach/reach-ui/blob/develop/packages/combobox/examples/token-input.js)

**Comment by tchak** [comment]
une fois que c'est fait, on peut supprimer select2 et jQuery de notre JavaScript (100k de chargement en moins sur la page d'accueil) cc @LeSim

**Comment by krichtof** [comment]
Petite coquille dans le lien pour l'exemple d'un token-input : https://github.com/reach/reach-ui/blob/develop/packages/combobox/examples/token-input.example.js

**Comment by krichtof** [comment]
#5903
